### PR TITLE
ST: ClusterOperator deployment via resources

### PR DESCRIPTION
**Pull request by Frawless**
### Type of change

- Enhancement / new feature
- Refactoring

### Description

ClusterOperator deployment for systemtests is using resources now. We are able to change properties before each test class very easily now. This change include some refactoring for _StrimziExtension_. 

### Checklist

- [x] Make sure all tests pass

### Systemtest results
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   SecurityST.testAutoRenewCaCertsTriggeredByAnno:212 » Timeout Timeout after 450...
[INFO] 
[ERROR] Tests run: 51, Failures: 0, Errors: 1, Skipped: 1
[INFO]

### CODE REVIEW

File: .checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -92,7 +92,7 @@
         </module>
         <module name="ClassDataAbstractionCoupling">
             <!-- default is 7 -->
-            <property name="max" value="20"/>
+            <property name="max" value="30"/>
```
[the remark this answers is not in the document]
Comment (Frawless) [reply .checkstyle/checkstyle.xml]: I think that 25 is enough, I raised to 30 just cause build errors. However it was at the start of refactor when I created CO as a CRD, not as fabric8 Deployment. 20 is maybe alright now, let me check it.

**Comment by Frawless** [comment]
@serrss  Well, we discuss this with @tombentley  few weeks ago. In general you are right. If someone do some changes in yamls and will not change it in ST it will be messy, however, using this deployment allow us to change CO configuration for each test very easily. Currently, it's not possible to do it without load yaml, parse it and apply, which is much time consuming that approach in this PR. We can create simple script/test which will compare values from CO yaml and COconstatns class during build. BTW I don't think this problem is solved somehow in Kafka resources.

File: systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java [hunk 2]
```diff
@@ -110,16 +111,17 @@
     static final long TEARDOWN_GLOBAL_WAIT = 10000;
 
     public static KubeClusterResource cluster = new KubeClusterResource();
-    protected static DefaultKubernetesClient client = new DefaultKubernetesClient();
+    protected static DefaultOpenShiftClient client = new DefaultOpenShiftClient();
```
Comment (tombentley) [inline systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java]: Why was this needed?

File: systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java [hunk 3]
```diff
@@ -110,16 +111,17 @@
...
     static KubeClient<?> kubeClient = cluster.client();
 
     Resources resources;
+    static Resources clusterOperatorResources;
```
Comment (tombentley) [inline systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java]: Doing this we will have two `Resources` floating around. The cluster one and one used in each test. That kind of makes sense, I suppose since the one is "scoped" to the whole class and the other to the test method. But I wonder then if `clusterOperatorResources` is the best name. after it doesn't have to be just used for the CO. Would `testClassResources` be better?

File: systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java [hunk 4]
```diff
@@ -391,28 +393,29 @@ public String  getInitContainerImageName(String podName) {
...
         collectLogs();
         LOGGER.info("Deleting resources after the test");
         resources.deleteResources();
         resources = null;
     }
 
+    protected static void deleteClusterOperatorResources() {
+        LOGGER.info("Deleting cluster operator resources after the test");
+        clusterOperatorResources.deleteResources();
+        clusterOperatorResources = null;
```
Comment (tombentley) [inline systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java]: Necessary?

File: systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java [hunk 5]
```diff
@@ -1054,4 +1057,33 @@ void waitForDeletion(long time, String namespace) throws Exception {
             throw new Exception("There are some unexpected pods! Cleanup is not finished properly!" + nonTerminated);
         }
     }
+
+    static void applyRoleBindings(String namespace, String clientNamespace) {
```
Comment (tombentley) [inline systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java]: Javadoc this one please!

File: systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java (hunk 3 above)
Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java]: Good point, I will change it

File: systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java (hunk 2 above)
Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java]: I think it's not necessary now. I did some experiments with RoleBindings, because I wasn't successful with deployment. I will put there `DefaultKubernetesClient` back.

File: systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java (hunk 4 above)
Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java]: You think that it's redundant because resources will be deleted by GC and pods by openshift during namespace deletion?

Comment (tombentley) [reply systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java]: I think so.

File: systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java [hunk 6]
```diff
@@ -1054,4 +1050,33 @@ void waitForDeletion(long time, String namespace) throws Exception {
...
+        testClassResources.kubernetesRoleBinding("../install/cluster-operator/032-RoleBinding-strimzi-cluster-operator-topic-operator-delegation.yaml", namespace, clientNamespace);
+    }
+
+    @BeforeEach
+    void setTestName(TestInfo testInfo) {
+        testName = testInfo.getTestMethod().get().getName();
+    }
+
+    @BeforeAll
+    static void setTestClassName(TestInfo testInfo) {
```
Comment (serrss) [inline systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java]: Maybe we need to update method name after adding logic for CO

Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java]: I agree, good point.
